### PR TITLE
Refactor Base\FunctionToStatic to configurable rule.

### PR DESCRIPTION
**Pull request by bbrala**
## Description
This one will be needed a lot in Drupal 10. By making it configurable before starting on d10 to d11 we save us a lot of time.

Want to use this one as an example d10 -> d11 rector rules. And perhaps as an example for a configurable rule with version in it.

## To Test
- Add steps to test this feature

## Drupal.org issue
Provide a link to the issue from https://www.drupal.org/project/rector/issues. If no issue exists, please create one and link to this PR.

### CODE REVIEW

File: src/Rector/ValueObject/FunctionToStaticConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace DrupalRector\Rector\ValueObject;
+
+class FunctionToStaticConfiguration {
+
+    protected string $deprecatedFunctionName = 'file_directory_os_temp';
+
+    protected string $className = 'Drupal\Component\FileSystem\FileSystem';
+
+    protected string $methodName = 'getOsTemporaryDirectory';
```
Comment (mglaman) [inline src/Rector/ValueObject/FunctionToStaticConfiguration.php]: Should this defaults exist?

**Review by mglaman — DISMISSED** [review]
The config shouldn't have default values for the properties, should it?

File: src/Rector/ValueObject/FunctionToStaticConfiguration.php (hunk 1 above)
Comment (bbrala) [reply src/Rector/ValueObject/FunctionToStaticConfiguration.php]: yeah it shouldnt, those leaked... will fix.

Comment (bbrala) [reply src/Rector/ValueObject/FunctionToStaticConfiguration.php]: Removed the default from the confguration.

**Review by agentrickard — APPROVED** [review]

**Pull request merged**